src/comodo/drakeSimulator/utils.py

```diff
@@ -0,0 +1,209 @@
+# some useful functions to load robot models
+# into the drake simulator
+
+import logging
+import os
+import sys
+import tempfile
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+import meshio
+import numpy as np
+from odio_urdf import *
+from pydrake.geometry import (
+    AddCompliantHydroelasticProperties,
+    AddContactMaterial,
+    AddRigidHydroelasticProperties,
+)
+from pydrake.geometry import Box as BoxDrake
+from pydrake.geometry import HalfSpace, ProximityProperties
+from pydrake.math import RigidTransform
+from pydrake.multibody.plant import CoulombFriction
+from tqdm import tqdm
+
+
+class DrakeURDFHelper:
+    """A simple class to make a given URDF drake compatible"""
+
+    def __init__(self):
+        pass
+
+    def load_urdf(self, urdf_path=None, urdf_string=None, mesh_path=None):
+        # read the urdf
+        self.urdf_path = str(urdf_path)
+        # save the string temporarily to parse the xml
+        # TODO: This function blows up if not careful during an optimisation
+        if urdf_string is not None:
+            tmpfile = tempfile.NamedTemporaryFile(mode="w+")
+            tmpfile.write(urdf_string)
+            self.urdf_path = tmpfile.name
+
+        self.mesh_path = mesh_path
+        self.urdf_out_path = os.getcwd() + "/urdfs"
+
+        # load the xml from the urdf_string
+        tree = ET.parse(self.urdf_path)
+        self.root = tree.getroot()
+
+    def convert_meshes_to_obj(self):
+        """Currently drake only loads .obj files and not .stl files.
+        This is a simple routine to allow this conversion"""
+        # read the meshes and convert them from .stl to .obj so that they can be loaded to Drake
+        logging.info("Converting all the meshes")
+        for child in tqdm(
+            self.root.findall("./link/visual/geometry/mesh")
+            or self.root.findall("./link/collision/geometry/mesh")
+        ):
+            # extract mesh location and name associated with the urdf
+            path = child.attrib["filename"]
+            child_mesh_path = self.mesh_path + path.split("package://")[1]
+            # convert the mesh and save it in the same folder
+            child_mesh_name = child_mesh_path.replace("stl", "obj")
+            if not Path(child_mesh_name).is_file():
+                temp_mesh = meshio.read(child_mesh_path)
+                temp_mesh.write(child_mesh_name)
+            # replace the urdf to load .obj files
+            child.set("filename", path.replace("stl", "obj"))
+
+        # # if the same, will be ignored else converted
```

Comment:
Could you remove  these comments and open an issue if is an issue needed to be addressed